mrmustard/lab/abstract/transformation.py

```diff
@@ -195,6 +132,53 @@ def d_vector_dual(self) -> Optional[RealVector]:
             return -d
         return -math.matmul(Xdual, d)
 
+    def bargmann(self, numpy=False):
+        X, Y, d = self.XYd(allow_none=False)
+        if self.is_unitary:
+            A, B, C = bargmann.wigner_to_bargmann_U(
+                X if X is not None else math.identity(d.shape[-1], dtype=d.dtype),
```

Comment:
when we call `self.XYd(allow_none=False)` the `allow_none=False` option already creates the default `X`,`Y`, and `d`, so there's no need here to have `if X is not None else ...` and same below